include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1174,7 +1177,8 @@ __parallel_find_or(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPoli
                             for (auto __old = __found.load(); __comp(__found_local.load(), __old);
```

Comment:
Can we also reduce this to be a `fetch_min` / `fetch_max` operation depending on the tag?  It seems like both the find_first would be a `fetch_min`, and the find_last and find_or would be a `fetch_max`.  It needs more thought to confirm, but I think its OK.

Reply:
Also fixed at https://github.com/oneapi-src/oneDPL/pull/1644/commits/42eb498d1a113aebf2e0ea1ed712c8cbc8b06c3c

Reply:
We are still doing a `__found.load()` and `__comp` operation.  I think we can avoid this completely, and replace it with `fetch_min` and `fetch_max`.  This should be applicable even in the case of `__or_tag_check`, because `fetch_max` should be a fine replacement for `__atomic_compare` as written.  

It *is* possible that this is slightly slower though, I'm not sure.  
As written, we have 1 atomic read guarantee, and an optional atomic write if the comparison returns true.  My proposal would replace that system with a single atomic read+write in the form of `fetch_max`. 

If we did this I think we might also be able to remove the `_Compare` type altogether.

Reply:
All of the atomic operations have this clause in their description (including the ones you reverted to).  What it is saying is that these 64 bit data type operations are only supported for device which support 64 bit atomics.  
I think we should use what is fastest here, and if we want to we can do some check of 
`__queue.get_device().has(sycl::aspect::atomic64)` when we have a 64 bit type like the one done here in single pass scan https://github.com/oneapi-src/oneDPL/blob/71de137931a0f0e23d29fb8dd6ac43352a9b46ed/include/oneapi/dpl/experimental/kt/single_pass_scan.h#L329

Reply:
Sure, we able to implement this approach.
But the amount of required changes in this case is too much.
Let's think about this in separate PR.

Reply:
And yes, of course, I already seen this code and condition check.

Reply:
OK, that is fine to separate to another PR.

Just to be clear though, `fetch_min` and `fetch_max` impose no additional requirements as compared to the current PR.